Src/FluentAssertions/Collections/SubsequentOrderingAssertions.cs

```diff
@@ -6,7 +6,7 @@ namespace FluentAssertions.Collections;
 
 [DebuggerNonUserCode]
 public class SubsequentOrderingAssertions<T>
-    : SubsequentOrderingGenericCollectionAssertions<IEnumerable<T>, T, SubsequentOrderingAssertions<T>>
+    : SubsequentOrderingGenericCollectionAssertions<IEnumerable<T>, T>
```

Comment:
🤔 I wonder why we even need `SubsequentOrderingAssertions` _and_ a `SubsequentOrderingGenericCollectionAssertions`. Maybe it's a weird usage of generic types that won't be apparent until you remove it.
🏕️ There's another `SubsequentOrderingGenericCollectionAssertions` that is also not used. Maybe we can remove that one as well.

Reply:
You are right. I moved the code from `SubsequentOrderingGenericCollectionAssertions` to `SubsequentOrderingAssertions` and could completely delete the `SubsequentOrderingGenericCollectionAssertions` class without breaking any tests.